src/components/dialog/DialogDismissibleView.js

```diff
@@ -0,0 +1,228 @@
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import React, {PureComponent} from 'react';
+import {Animated, Easing, StyleSheet} from 'react-native';
+import {Constants} from '../../helpers';
+import View from '../view';
+import asPanViewConsumer from '../panningViews/asPanViewConsumer';
+import PanningProvider from '../panningViews/panningProvider';
+import PanResponderView from '../panningViews/panResponderView';
+
+const MAXIMUM_DRAGS_AFTER_SWIPE = 2;
+
+class DialogDismissibleView extends PureComponent {
+  static propTypes = {
+    /**
+     * The direction of the allowed pan (default is DOWN)
+     * Types: UP, DOWN, LEFT and RIGHT (using PanningProvider.Directions.###)
+     */
+    direction: PropTypes.oneOf(Object.values(PanningProvider.Directions)),
+    /**
+     * onDismiss callback
+     */
+    onDismiss: PropTypes.func,
+    /**
+     * Is this bottom, top or none (center)
+     */
+    alignment: PropTypes.shape(PropTypes.arrayOf[PropTypes.bool]),
+    /**
+     * The dialog`s container style
+     */
+    containerStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
+    /**
+     * Whether to show the dialog or not
+     */
+    visible: PropTypes.bool,
+  };
+
+  static defaultProps = {
+    direction: PanningProvider.Directions.DOWN,
+    onDismiss: _.noop,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.hiddenLocation = {};
+    this.swipe = {};
+    this.animatedValue = new Animated.Value(0);
+    this.state = {
+      visible: props.visible,
+      isAnimating: false,
```

Comment:
@ethanshar
`isAnimating` is not actually used, but when I tried to remove it the animation did not work. Is there a better solution?

Reply:
I know that the name of the value is not the important part.
This is my theory after working on it some more: the first render happens before the `onLayout` and if we don't use `setState` it does not know anything has changed (even if I put the `animatedValue` on the `state`).
What do you think?

Reply:
let's go over it together and investigate

Reply:
Added a TODO for this issue